src/os/open.zig

```diff
@@ -77,3 +81,131 @@ const OpenCommand = struct {
     child: std.process.Child,
     wait: bool = false,
 };
+
+/// Use `xdg-open` to open a URL using the default application.
+///
+/// Any output on stderr is logged as a warning in the application logs. Output
+/// on stdout is ignored.
+pub fn openUrlLinux(
+    alloc: Allocator,
+    url: [:0]const u8,
+) void {
+    // Make a copy of the URL so that we can use it in the thread without
+    // worrying about it getting freed by other threads.
+    const copy = alloc.dupe(u8, url) catch |err| {
+        log.warn("unable to copy URL before opening: {}", .{err});
+        return;
+    };
+    // Run `xdg-open` in a thread so that it never blocks the main thread, no
+    // matter how long it takes to execute.
+    const thread = std.Thread.spawn(.{}, _openUrlLinux, .{ alloc, copy }) catch |err| {
+        alloc.free(url);
+        log.warn("unable to start thread to launch url: {}", .{err});
+        return;
+    };
+    // Don't worry about the thread any more.
+    thread.detach();
+}
+
+fn _openUrlLinux(alloc: Allocator, url: []const u8) void {
+    defer alloc.free(url);
+
+    var exe = std.process.Child.init(
+        &.{ "xdg-open", url },
+        alloc,
+    );
+
+    // We're only interested in stderr
+    exe.stdin_behavior = .Ignore;
+    exe.stdout_behavior = .Ignore;
+    exe.stderr_behavior = .Pipe;
+
+    exe.spawn() catch |err| {
```

Comment:
Having bunch of `catch` in this is nasty and makes it so `err defer` can't be used (if we even wanted to). I recommend the following pattern instead:

```zig
fn openUrlLinuxThread(...) void {
    // Handle error once
    openUrlLinuxThreadError(...) catch |err| {}
}

fn openUrlLinuxThreadError(...) !void {
    // Return errors like normal, use errdefer.
}
```

Reply:
Fixed.